src/main/java/com/onarandombox/MultiversePortals/MultiversePortals.java

```diff
@@ -415,9 +416,37 @@ public FileConfiguration getMainConfig() {
     }
 
     public void reloadConfigs() {
-        this.portalManager.removeAll(false);
-        this.loadPortals();
+        this.reloadConfigs(true);
+    }
+
+    public void reloadConfigs(boolean reloadPortals) {
+        if (reloadPortals) {
+            this.portalManager.removeAll(false);
+            this.loadPortals();
+        }
+
+        PluginManager pm = this.getServer().getPluginManager();
+        boolean previousTeleportVehicles = MultiversePortals.TeleportVehicles;
+        boolean previousUseOnMove = MultiversePortals.UseOnMove;
+
         this.loadConfig();
+
+        if (MultiversePortals.TeleportVehicles != previousTeleportVehicles) {
+            if (MultiversePortals.TeleportVehicles) {
+                pm.registerEvents(new MVPVehicleListener(this), this);
+            } else {
+                VehicleMoveEvent.getHandlerList().unregister(this);
+            }
```

Comment:
Are you sure you're not suppose to pass our listener instance in here?

Reply:
Ahh, I see there are two versions of the method and this one removes all our listeners for that event.